spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkWriteOptions.java

```diff
@@ -53,4 +53,11 @@ private SparkWriteOptions() {
 
   // Controls whether to allow writing timestamps without zone info
   public static final String HANDLE_TIMESTAMP_WITHOUT_TIMEZONE = "handle-timestamp-without-timezone";
+
+  // Overrides the default distribution mode for a write operation
+  public static final String DISTRIBUTION_MODE = "distribution-mode";
+
+  // Controls whether to ignore the table sort order during a write operation
+  public static final String IGNORE_SORT_ORDER = "ignore-sort-order";
```

Comment:
What does this mean? Treat the table as unsorted? What about distribution, does that require overriding: `distribution-mode=none`? Looks like the current implementation of `SparkWriteConf.distributionMode()` does not check `ignoreSortOrder`.

It makes sense to me to have an option like this, so that someone writing a job can turn off automatic sort order and partitioning and can customize by adding a sort to the job instead. My main concern is that we are consistent and it may be easier to turn off required distribution and ordering together rather than separately. For example, if you set `ignore-sort-order=true` but the table's distribution mode is still `range`, then `SparkDistributionAndOrderingUtil` will still build a global sort using the table's sort order: https://github.com/apache/iceberg/pull/3461/files#diff-acf71a9ace396ea8a334c93bbf55629f493eb830124ef5e6a41a5a6a4e200af6R53

Maybe this should be `ignore-table-distribution-and-sort` instead?

Reply:
I went back and forth on this one and was not sure whether I am overthinking. I did not have a good name so kept it as is. Let me think more on this. I think I like having only one property to disable both. That's easier.

Reply:
What about `use-table-write-config`, with a default to `true`?

I like that because it is phrased positively, rather than using true to turn off a setting. It also reuses "table" and "write" from our DDL: `ALTER TABLE ... WRITE ...`. Config includes both sort order and distribution, which are set using the same DDL statement.

Reply:
+1

Reply:
After thinking a little bit more, I am afraid `write-config` is too abstract and may be interpreted like all write properties should be ignored. I went with `use-table-distribution-and-ordering` for now. Let me know what you think.